testbed/tests/app/test_desktop.py

```diff
@@ -180,106 +182,269 @@ async def test_menu_minimize(app, app_probe):
         window1.close()
 
 
-async def test_full_screen(app, app_probe):
-    """Window can be made full screen"""
-    window1 = toga.Window("Test Window 1", position=(150, 150), size=(200, 200))
-    window2 = toga.Window("Test Window 2", position=(400, 150), size=(200, 200))
-
+async def test_presentation_mode(app, app_probe, main_window):
+    """The app can enter presentation mode."""
     try:
-        window1.content = toga.Box(style=Pack(background_color=REBECCAPURPLE))
-        window2.content = toga.Box(style=Pack(background_color=CORNFLOWERBLUE))
-        window1_probe = window_probe(app, window1)
-        window2_probe = window_probe(app, window2)
+        window_information_list = list()
+        windows_list = list()
+        for i in range(len(app.screens)):
+            window = toga.Window(
+                title=f"Test Window {i}",
+                position=(150 + (10 * i), 150 + (10 * i)),
+                size=(200, 200),
+            )
+            r = random.randint(0, 255)
+            g = random.randint(0, 255)
+            b = random.randint(0, 255)
+            window.content = toga.Box(
+                style=Pack(background_color=f"#{r:02X}{g:02X}{b:02X}")
+            )
+            window.show()
+            # Add delay for gtk to show the windows
+            await app_probe.redraw(f"Test Window {i} is visible", delay=0.5)
+
+            window_information = dict()
+            window_information["window"] = window
+            window_information["window_probe"] = window_probe(app, window)
+            window_information["initial_content_size"] = window_information[
+                "window_probe"
+            ].presentation_content_size
+
+            window_information_list.append(window_information)
+            windows_list.append(window)
+
+        screen_window_dict = dict()
+        for window, screen in zip(windows_list, app.screens):
+            screen_window_dict[screen] = window
+
+        # Enter presentation mode with a screen-window dict via the app
+        app.enter_presentation_mode(screen_window_dict)
+        # Add delay for gtk to show the windows
```

Comment:
No need to mention specific backends; the *generic* purpose is enough ("delay ensure windows are visible after animation")

Reply:
Thanks. Changed it.